app/models/repository.rb

```diff
@@ -165,6 +175,18 @@ def relative_path(path)
     path
   end
 
+  def required_disk_storage
+    if scm.storage_available?
+      if storage_updated_at.nil? ||
+         storage_updated_at < 12.hours.ago
+
+        Delayed::Job.enqueue ::Scm::StorageUpdaterJob.new(self)
+      end
+
+      required_storage_bytes
```

Comment:
That is exactly right and the reason this value may be externally updated (through the Sys api) and we may have to update this value regularly. I chose not to do this simply because the exact requirement on timeliness wasn't clear, and  in the current state, all those possibilities remain easy to implement.